### ISSUE TITLE
[Feature] Expand apply_token_bitmask_inplace_cpu to support bfloat16 logits.

### ISSUE THREAD

**Issue by jmaksymc** [mlc-ai/xgrammar#447]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The latest version does not have this feature.
- [x] 3. I have filled all the required information below.
- [x] 4. The feature request is in English.

### requested feature

Expand apply_token_bitmask_inplace_cpu to support not only float32 logits. Add support for logits in bfloat16.

### Motivation

In xgrammar v0.1.6 apply_token_bitmask_inplace_cpu used to support logits in not only float32 precision. Since then this kernel has been rewritten in cpp and supports only float32 logits. Many models run with bfloat16 logits, especially on CPU, thus such support would be very helpful.

### Alternatives

Downgrading xgrammar version to 0.1.6, upcasting the logits to float32 before applying the bitmask.

**Comment by cosmicallyrun** [comment]
Support for this! bfloat16 would be very helpful.

**Comment by Seven-Streams** [comment]
Sorry for the late response, I'll try to support it soon.

**Comment by Seven-Streams** [comment]
Completed in #469, and the issue will be closed for now. Please feel free to reopen it if you like.
